tests/test_flow/test_reload.py

```diff
@@ -40,216 +40,242 @@ def _write_module(name, contents):
     sys.dont_write_bytecode = dont_write_bytecode
 
 
+@pytest.fixture(params=["reload", "reloading"])
+def reload_method_name(request):
+    return request.param
+
+
 @pytest.fixture
-def write_flow_module(write_module):
-    def _write_flow_module(x):
-        write_module(
-            "flow_module",
-            f"""
+def flow_tester(reload_method_name, tmp_path, write_module):
+    class FlowTester:
+        """
+        A helper class for testing dynamically written flows.
+
+        If the reload method is reloading(), whenever the method is called, we get a
+        new copy of the flow with reloaded modules. We need to verify that the
+        entities of the original flow instance give the same values when computed as
+        before the reload.
+        """
+
+        def __init__(self):
+            self.flow = None
+            self.expected_results_by_flow = {}
+
+        def write_tester_flow(self, body):
```

Comment:
nit: this method might be clearer as `write_flow_module`; that makes it a bit clearer how it relates to `write_module`, and of course the word `tester` is a bit redundant here.